framework/src/modules/nft/method.ts

```diff
@@ -195,19 +195,39 @@ export class NFTMethod extends BaseMethod {
 		nftID: Buffer,
 	): Promise<boolean> {
 		const nftChainID = this.getChainID(nftID);
-		if (nftChainID.equals(this._config.ownChainID)) {
+		if (this._isNFTFromOwnChain(nftChainID)) {
 			return true;
 		}
 
 		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);
-		const supportForAllKeysExists = await supportedNFTsStore.has(
-			methodContext,
-			ALL_SUPPORTED_NFTS_KEY,
-		);
-		if (supportForAllKeysExists) {
+		if (await this._isAllNFTsSupported(supportedNFTsStore, methodContext)) {
+			return true;
+		}
+
+		if (await this._isNFTChainSupported(supportedNFTsStore, methodContext, nftChainID, nftID)) {
 			return true;
 		}
 
+		return false;
+	}
+
+	private _isNFTFromOwnChain(nftChainID: Buffer): boolean {
+		return nftChainID.equals(this._config.ownChainID);
+	}
+
+	private async _isAllNFTsSupported(
+		supportedNFTsStore: SupportedNFTsStore,
+		methodContext: ImmutableMethodContext,
+	): Promise<boolean> {
+		return supportedNFTsStore.has(methodContext, ALL_SUPPORTED_NFTS_KEY);
+	}
+
+	private async _isNFTChainSupported(
+		supportedNFTsStore: SupportedNFTsStore,
+		methodContext: ImmutableMethodContext,
+		nftChainID: Buffer,
+		nftID: Buffer,
+	): Promise<boolean> {
```

Comment:
Unless these functions can be reused in other place(s), isolating them just creates an extra overhead of function calls and return (with an additional boolean return)

Reply:
I think this is not a show stopper but rather a style or design issue. If there is overhead, it is negligible. IMHO ie

Reply:
The overhead is of'course not significant but the question probably is why refactor existing code to something less efficient unless there is a clear benefit ?

Reply:
I only refactored based on our last discussion where you requested a refactor of the function. Otherwise wouldn't have. Anyway, I have no strong opinions on making functions long.

Reply:
It was probably a miscommunication, because I only requested the refactor of endpoint but not the method. I also don't have a strong opinion/preference on single vs multiple functions but in this particular case I'd avoid changes or deviation from the LIP unless strictly necessary as it is an update to release/6.1.0